src/components/username-manager.ts

```diff
@@ -96,7 +121,7 @@ export default class UsernameManager extends BotComponent {
             );
             await member.setNickname(new_name);
         } else {
-            return;
+            await this.check_herald(member);
```

Comment:
We may as well do this check always, not just if the user is detected to not have troublesome unicode in their name

Reply:
Yeah, I was wondering that too, just wanted to change the code as little as possible. But sure, fixed :)